### PR TITLE
Add additional tests to JS transformer

### PR DESCRIPTION
**Pull request by colinking**
## Description

Extends the test coverage on the JS transformer. Adds support for serializing parameters as strings (instead of objects) if only the type is set. Fixes quote formatting s.t. single/double quotes are chosen to minimize string length.

## Test plan

Unit tests pass.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #567** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/567" target="_blank">Graphite</a> 
    * **PR #568** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/568" target="_blank">Graphite</a> 
      * **PR #571** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/571" target="_blank">Graphite</a> 
        * **PR #572** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/572" target="_blank">Graphite</a> 
          * **PR #580** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/580" target="_blank">Graphite</a> 
            * **PR #581** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/581" target="_blank">Graphite</a> 
              * **PR #582** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/582" target="_blank">Graphite</a> 
                * **PR #590** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/590" target="_blank">Graphite</a> 
                  * **PR #591** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/591" target="_blank">Graphite</a>  👈
                    * **PR #592** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/592" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/lib/591?utm_source=stack-comment).

File: pkg/runtime/javascript/javascript.go [hunk 1]
```diff
@@ -576,39 +576,46 @@ func (r Runtime) SupportsLocalExecution() bool {
...
+				matches := airplaneErrorRegex.FindStringSubmatch(string(ee.Stderr))
+				if len(matches) >= 2 {
+					errMsg := matches[1]
+					return errors.Errorf("editing task at %q (re-run with --debug for more context): %s", path, errMsg)
+				}
 			}
+			return errors.Wrapf(err, "editing task at %q (re-run with --debug for more context)", path)
 		}
-		return errors.Wrapf(err, "editing task at %q (re-run with --debug for more context)", path)
+	case ".yaml", ".yml", ".json":
```
Comment (ameyakhare) [inline pkg/runtime/javascript/javascript.go:614]: when would this case hit?

File: pkg/runtime/javascript/transformer/transformer.ts [hunk 2]
```diff
@@ -99,19 +98,25 @@ const buildTaskConfig = (def: any): ExpressionKind => {
...
-        rest.default = asRecastNode(buildParamValue(rest.default, rest.type));
-      }
-      // Convert option parameter values from JSON to JS values.
-      // Options without a label can be serialized as the value directly.
-      if (rest.options) {
-        rest.options = rest.options.map((option) => {
-          option.value = asRecastNode(buildParamValue(option.value, rest.type));
-          return option.label ? option : option.value;
-        });
+      if (Object.keys(rest).length === 1) {
```
Comment (ameyakhare) [inline pkg/runtime/javascript/transformer/transformer.ts:101]: nice

**Review by ameyakhare — APPROVED** [review]

File: pkg/runtime/javascript/javascript.go (hunk 1 above)
Comment (colinking) [reply pkg/runtime/javascript/javascript.go:613]: I added this in a future PR: https://github.com/airplanedev/lib/pull/592

**Pull request merged**